EIPS/eip-approval_by_amount.md

```diff
@@ -0,0 +1,122 @@
+     * @notice Returns the amount allocated to `operator` approved to transfer ``account``'s tokens, according to `id`.
+     */
+    function allowance(address account, address operator, uint256 id) external view returns (uint256);
+    
+}
+```
+
+The `_allowances` mapping MAY be implemented as `internal` or `private`.
+
+The `approve(address operator, uint256 id, uint256 amount)` function MAY be implemented as `public` or `external` and it is RECOMMENDED implements `private` `_approve(address owner, address operator, uint256 id, uint256 amount)` function, which is responsible for performing the approval.
+
+The `allowance(address account, address operator, uint256 id)` function MAY be implemented as `public` or `external` for visibility and MUST be implemented as `view`.
+
+Every contract implementing the `ERC1155ApprovalByAmount` extension MUST implement `_checkApprovalForBatch(address from, address to, uint256[] memory ids, uint256[] memory amounts)` function in some way. This function is necessary to manage
+batch transfers subtracting the amount of tokens transferred.
+
+In same way, `safe(Batch)TrasferFrom` overrided function from `ERC1155` standard it is also important to change its implementation:
+
+- In `safeTransferFrom` MUST be added to require an extra condition: `|| allowance(from, _msgSender(), id)`.
+- In `safeTransferFrom` MUST be subtracted from `_allowances` mapping the transferred amount of tokens: `_allowances[from][_msgSender()][id] -= amount`.
+- In `safeBatchTransferFrom` MUST be added to require an extra condition that checks if the `allowance` of all `ids` have the approved `amounts` (See `_checkApprovalForBatch` function reference implementation)
+
+The `ApprovalByAmount` event MUST be emitted when a certain number of tokens are approved.
+
+The `supportsInterface` method MUST return `true` when called with `0x1be07d74`.
+
+## Rationale
+
+The chosen name resonates with the purpose of its existence. Users can approve their tokens by id and amounts to `operators`.
+
+The [ERC-20](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-20.md) standard name patterns, were used to simplify the understanding of how the interface works due to similarity with how [ERC-20](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-20.md) works with approvals. This decision was made after been discussed at [Ethereum Magicians forum](https://ethereum-magicians.org/t/working-draft-new-extension-for-erc1155).
+
+## Backwards Compatibility
+
+This standard is compatible with current [ERC-1155](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1155.md) standards.
+
+## Test Cases
+
+Test cases are available in the reference implementation [here](https://github.com/ivanmmurciaua/ERC1155ApprovalByAmount/blob/main/test/erc1155approvalbyamount.js).
+
+## Reference Implementation
+
+The reference implementation can be found [here](https://github.com/ivanmmurciaua/ERC1155ApprovalByAmount/blob/main/contracts/ERC1155ApprovalByAmount.sol).
+
```

Comment:
```suggestion
```
External links not allowed. You can include the test cases inline or as a file in `../assets/eip-5216/` and link to it here.  Alternatively, this is an optional section so you can leave it out for now.

Reply:
I will make a new commit with assets for your revision. Thanks!